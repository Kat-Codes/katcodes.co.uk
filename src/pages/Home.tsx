import React, { FunctionComponent } from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { device } from "../utils/breakpoints";
import Title from "../atoms/Title";
import Button from "../atoms/Button";

const Home: FunctionComponent = () => (
  <TitleContainer>
    <Title id="homepage-title">
      Good <Time />, I'm Katie
    </Title>
    <p>I'm a UK based Software Engineer working at Spotify ðŸŽµ</p>
    <Link to="/contact">
      <Button data-qa-id="hello-btn">Say hello</Button>
    </Link>
  </TitleContainer>
);

const Time = () => {
  let message;
  const hour = new Date().getHours();

  if (hour < 12) {
    message = "morning";
  } else if (hour < 18) {
    message = "afternoon";
  } else {
    message = "evening";
  }
  return <StyledTime>{message}</StyledTime>;
};

const StyledTime = styled.span`
  text-decoration: underline;
`;

const TitleContainer = styled.div`
  width: 100%;
  padding: 125px 0px;

  > * {
    margin: 16px 0px;
  }

  ${device.tablet} {
    width: 50%;
  }
`;

export default Home;
