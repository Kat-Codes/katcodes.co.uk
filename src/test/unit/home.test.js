import Home from '../../pages/Home';
import React from 'react';
import { renderWith } from './testUtils';

describe('Home page', () => {
    test('should render the home page as expected', () => {
        const component = renderWith(<Home />)
        expect(component).toMatchSnapshot();
    });
});
