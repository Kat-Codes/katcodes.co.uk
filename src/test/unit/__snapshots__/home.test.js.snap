// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render the home page as expected 1`] = `
<div
  className="sc-hKwDye jQoBYu"
>
  <h1
    className="sc-bdvvtL fOrCqx"
    id="homepage-title"
  >
    Good 
    <span
      className="sc-dkPtRN dmoJZt"
    >
      afternoon
    </span>
    , I'm Katie
  </h1>
  <p>
    I'm a UK based Software Engineer working at Spotify ðŸŽµ
  </p>
  <a
    href="/contact"
    onClick={[Function]}
  >
    <button
      className="sc-gsDKAQ fhvAAj"
      data-qa-id="hello-btn"
    >
      Say hello
    </button>
  </a>
</div>
`;
